erpnext/projects/doctype/timesheet/timesheet.py

```diff
@@ -136,10 +136,15 @@ def validate_dates(self):
 
 	def validate_time_logs(self):
 		for data in self.get('time_logs'):
+			self.set_to_time(data)
 			self.validate_overlap(data)
 			self.set_project(data)
 			self.validate_project(data)
 
+	def set_to_time(self, data):
+		if data.from_time and data.hours and not data.to_time == add_to_date(data.from_time, hours=data.hours):
```

Comment:
This condition will almost never be true? Because `to_time` will be string and `add_to_date` returns datetime object. Maybe use `get_datetime`?